src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/RuntimeODP.java

```diff
@@ -0,0 +1,396 @@
+                                  String splitRow, String packageSize, String msServ, String systemID,
+                                  String lgrp) throws IOException, InterruptedException {
+    ODPActions.getODPProperties();
+    ODPActions.enterConnectionProperties(client, sysnr, asHost, dsName, gcsPath, splitRow, packageSize,
+                                         msServ, systemID, lgrp);
+  }
+
+  @When("Username and Password is provided")
+  public void usernameAndPasswordIsProvided() throws IOException {
+    ODPActions.enterUserNamePassword(CDAPUtils.getPluginProp("AUTH_USERNAME"),
+                                     CDAPUtils.getPluginProp("AUTH_PASSWORD"));
+  }
+
+  @When("Run one Mode is Sync mode")
+  public void runOneModeIsSyncMode() {
+    ODPActions.selectSync();
+  }
+
+  @Then("Validate the Schema created")
+  public void validateTheSchemaCreated() throws InterruptedException {
+    ODPActions.getSchema();
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 20000);
+    wait.until(ExpectedConditions.numberOfElementsToBeMoreThan
+      (By.xpath("//*[@placeholder=\"Field name\"]"), 2));
+  }
+
+  @Then("Close the ODP Properties")
+  public void closeTheODPProperties() {
+    ODPActions.closeButton();
+  }
+
+  @Then("Delete the existing Odp table {string} in bigquery")
+  public void deleteTheExistingOdpTableInBigquery(String table) throws IOException, InterruptedException {
+    gcpClient.dropBqQuery(CDAPUtils.getPluginProp(table));
+    BeforeActions.scenario.write("Table Deleted Successfully");
+  }
+
+  @Then("link source and target")
+  public void linkSourceAndTarget() {
+    SeleniumHelper.dragAndDrop(ODPLocators.fromODP, CdfStudioLocators.toBigQiery);
+  }
+
+  @Then("Delete all existing records in datasource")
+  public void deleteAllExistingRecordsInDatasource() throws JCoException, IOException {
+    Properties connection = readPropertyODP();
+    sapProps = SAPProperties.getDefault(connection);
+    errorCapture = new ErrorCapture(exceptionUtils);
+    sapAdapterImpl = new SAPAdapterImpl(errorCapture, connection);
+    Map opProps = new HashMap<>();
+    opProps.put("RFC", "ZTEST_DATA_DEL_ACDOCA");
+    opProps.put("autoCommit", "true");
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      ObjectNode objectNode = mapper.createObjectNode();
+      objectNode.put("I_ALL", "X");
+      JsonNode response = sapAdapterImpl.executeRFC(objectNode.toString(), opProps,
+                                                    StringUtils.EMPTY, StringUtils.EMPTY);
+    } catch (Exception e) {
+      throw SystemException.throwException(e.getMessage(), e);
+    }
+  }
+
+  public Properties readPropertyODP() throws IOException {
+    Properties prop = new Properties();
+    InputStream input;
+    Properties connection = new SAPProperties();
+    input = new FileInputStream("src/e2e-test/resources/Google_SAP_Connection.properties");
+    prop.load(input);
```

Comment:
can we load property file once instead on every call?

Reply:
done